libvast/src/segment_store.cpp

```diff
@@ -351,8 +353,10 @@ caf::error segment_store::flush() {
     return err;
   // Keep new segment in the cache.
   cache_.emplace(seg.id(), seg);
+  [[maybe_unused]] const auto trimmed_filename
+    = filename.string().substr(0, filename.string().size() - 3);
   VAST_DEBUG("{} wrote new segment to {}", detail::pretty_type_name(this),
-             filename.trim(-3));
```

Comment:
The function `path::trim` operated not on characters but instead on path components. Check the docs here: https://github.com/tenzir/vast/blob/master/libvast/vast/path.hpp#L105. Here, `-3` means calling `filesystem::path::parent_path` three times.

The logic should be easy to factor into a free function in namespace detail. However, I don't recall that we use this function aside from prettifying logging output. And in this particular instance, I wonder whether simply calling `filename.parent_path()` is what we want.

Reply:
Oops, thanks for pointing that out -- definitely an unintended change in behavior.

I just switched to using `filename.parent_path()` here as it's arguably better. If you or @dominiklohmann feel strongly, we can just `filename.parent_path().parent_path().parent_path()` to mimic the old behavior.

Reply:
This works in my opinion.

It's also a debug log statement. We don't have a policy of, say, printing all paths relative to the `vast.db` dir in log statements. It's something to consider for the future and only on log levels of INFO and more severe.